judo-ui-react/src/main/resources/actor/src/containers/components/table.tsx.hbs

```diff
@@ -327,536 +270,128 @@ export function {{ componentName table }}(props: {{ componentName table }}Props)
-        getRowClassName={ (params: GridRowClassNameParams) => {
-          {{# if (stringValueIsTrue useTableRowHighlighting) }}
-          const customStyle = rowStylings.find(r => r.condition(params));
-          if (customStyle) {
-            return customStyle.name;
-          }
-          {{/ if }}
-          return params.indexRelativeToCurrentPage % 2 === 0 ? 'even' : 'odd';
-        } }
-        columns={effectiveTableColumns}
-        {{# if isMUILicensePlanPro }}
-        onColumnOrderChange={onColumnsChanged}
-        onColumnWidthChange={onColumnsChanged}
+        {{# if (getRefreshActionDefinitionForTable table) }}
+        maskAction={actions.get{{ firstToUpper table.relationName }}Mask}
         {{/ if }}
-        {{# unless table.isSelectorTable }}
-        disableRowSelectionOnClick
-        {{/ unless }}
-        {{# if table.isSelectorTable }}
-        isRowSelectable={handleIsRowSelectable}
-        hideFooterSelectedRowCount={ !{{ boolValue table.allowSelectMultiple }} }
+        enabledByName='{{ table.enabledBy.name }}'
+        dataElement={ ownerData?.{{ table.dataElement.name }} }
+        relationName='{{ table.relationName }}'
+        {{# if container.isSelector }}
+        selectionDiff={selectionDiff}
+        setSelectionDiff={setSelectionDiff}
         {{/ if }}
-        {{# if (tableHasSelectorColumn table) }}
-        checkboxSelection
-        rowSelectionModel={selectionModel}
-        onRowSelectionModelChange={handleOnSelection}
+        >
+        </EagerTable>
+      {{ else }}
+        <LazyTable<{{# with (getReferenceClassType table) as |classType| }}{{# if classType.isMapped }}{{ classDataName (getReferenceClassType table) 'Stored' }}{{ else }}{{ classDataName (getReferenceClassType table) '' }}{{/ if }}{{/ with }}, {{ classDataName (getReferenceClassType table) 'Stored' }}, {{ classDataName (getReferenceClassType table) 'QueryCustomizer' }}>
+        uniqueId={`{{ createId table }}-${uniqueId}`}
+        {{# if (stringValueIsTrue useTableRowHighlighting) }}
+        rowHighlightingHookInterfaceKey={`(&(${OBJECTCLASS}=${TABLE_ROW_HIGHLIGHTING_HOOK_INTERFACE_KEY})(component={{~ componentName table ~}}))`}
         {{/ if }}
-        keepNonExistentRowsSelected
+        {{# with (getReferenceClassType table) as |classType| }}
+        dataProp={ {{# if table.isSelectorTable }}[]{{ else }}{{# if table.isEager }}ownerData?.{{ table.dataElement.name }} || []{{ else }}[]{{/ if }}{{/ if }} }
+        {{/ with }}
+        defaultSortParamsForTable={ {{{ getDefaultSortParamsForTable table }}} }
+        tablePageLimit={ {{ calculateTablePageLimit table }} }
+        tableColumns={columns}
+        tableRowActions={rowActions}
+        tableFilterOptions={filterOptions}
+        mask={ '{{ getMaskForTable table }}' }
+        isOwnerLoading={isOwnerLoading}
+        validationError={validationError}
+        actions={actions}
+        allowSelectMultiple={ {{ boolValue table.allowSelectMultiple }} }
```

Comment:
I believe every prop which is filled with parameters created by `boolValue` shuold be replacable with a handlebars `{{# if ` if the boo props are optional, this could save 4-6 prop declarations here if we conditionally fill the props with `true` if `boolValue` yields `true`.

Reply:
[f51c46b](https://github.com/BlackBeltTechnology/judo-ui-react-template/pull/365/commits/f51c46bc6ce006871f020c8d52167c195794f251)